### ISSUE TITLE
https_proxy support

### ISSUE THREAD

**Issue by McSlow** [puppetlabs/puppetlabs-image_build#24]
Hi,
currently evaluating this module, but i'm stuck when it tries to reach a ressource using https:

invocation (it's the hello world example):

    puppet docker build --apt-proxy $http_proxy --http-proxy $http_proxy

...downloads all the apt stuff until it comes to the point where it requests a https ressource ( triggered by : .. &&     wget https://apt.puppetlabs.com/puppetlabs-release-pc1-"$CODENAME".deb &&.. )

    --2017-02-10 12:57:37--  https://apt.puppetlabs.com/puppetlabs-release-pc1-xenial.deb
    Resolving apt.puppetlabs.com (apt.puppetlabs.com)... 192.155.89.90, 2600:3c03::f03c:91ff:fedb:6b1d

it fails there as the https_proxy isn't set. An option --https-proxy isn't available yet.

Am I doing something wrong?

version.:

    /etc/puppet/modules
    └── puppetlabs-image_build (v0.1.2)

tried to get around this by setting ```--env https_proxy=$https_proxy``` but same result

**Comment by garethr** [comment]
@McSlow this looks like a bug, or rather a partially complete feature. The correct fix as you mention is to add a new `https-proxy` option. Thanks for reporting, this is definitely something we'd like to support in the future.

**Comment by McSlow** [comment]
@garethr I already prepared a tested PR which adds --https-proxy, will pull it on Monday :)

btw.: i also found out why my "--env" workaround didnt work. currently all env keys will be .upcase'd , so adding an --env https_proxy=<https_proxy> will result in an HTTPS_PROXY environment. wget ( which is used there) doesn't take this into account and needs the lowercase variant. http_proxy currently has the same issue as it sets an HTTP_PROXY build_env param. I fixed that, too.
